libcudacxx/include/cuda/std/__memory/addressof.h

```diff
@@ -21,22 +21,30 @@
 #  pragma system_header
 #endif // no system header
 
+_CCCL_DIAG_PUSH
+_CCCL_DIAG_SUPPRESS_MSVC(4312) // warning C4312: 'type cast': conversion from '_Tp' to '_Tp *' of greater size
+
 _LIBCUDACXX_BEGIN_NAMESPACE_STD
 
-// addressof
-// NVCXX has the builtin defined but did not mark it as supported
-#if defined(_CCCL_BUILTIN_ADDRESSOF)
+#if _CCCL_HAS_BUILTIN_STD_ADDRESSOF()
+
+// The compiler treats ::std::addressof as a builtin function so it does not need to be
+// instantiated and will be compiled away even at -O0.
+using ::std::addressof;
+
+#elif defined(_CCCL_BUILTIN_ADDRESSOF) // ^^^ _CCCL_HAS_BUILTIN_STD_ADDRESSOF() ^^^ /
+                                       // vvv _CCCL_BUILTIN_ADDRESSOF vvv
 
 template <class _Tp>
-_LIBCUDACXX_HIDE_FROM_ABI constexpr _LIBCUDACXX_NO_CFI _Tp* addressof(_Tp& __x) noexcept
+[[nodiscard]] _LIBCUDACXX_HIDE_FROM_ABI _LIBCUDACXX_NO_CFI constexpr _Tp* addressof(_Tp& __x) noexcept
```

Comment:
This is missing `_CCCL_INTRINSIC` applies throughout

Reply:
`addressof` had `_CCCL_INTRINSIC` in [b73b2ff1857ffaf3fc838b407f71a2fd7ad7f6e7] but i had to back it out. msvc didn't like it.